dnf-behave-tests/dnf/plugins-core/config-manager-addrepo.feature

```diff
@@ -0,0 +1,437 @@
+    And file "/etc/yum.repos.d/test.repo" contents is
+        """
+        [test]
+        name=repository file
+        enabled=0
+        baseurl=http://something.com/os/
+        """
+
+
+Scenario: "addrepo" from repofile, source file without .repo extension, adding the .repo extension to the destination filename
+ Given I create file "/{context.dnf.tempdir}/tmp/test" with
+        """
+        [test]
+        name=repository file
+        enabled=0
+        baseurl=http://something.com/os/
+        """
+    And I execute dnf with args "config-manager addrepo --from-repofile={context.dnf.tempdir}/tmp/test"
+   Then the exit code is 0
+    And file "/etc/yum.repos.d/test.repo" contents is
+        """
+        [test]
+        name=repository file
+        enabled=0
+        baseurl=http://something.com/os/
+        """
+
+
+Scenario: test "addrepo" from repofile with user defined destination file name (--save-filename=)
+ Given I create file "/{context.dnf.tempdir}/tmp/original.repo" with
+        """
+        [test]
+        name=repository file
+        enabled=0
+        baseurl=http://something.com/os/
+        """
+   When I execute dnf with args "config-manager addrepo --from-repofile={context.dnf.tempdir}/tmp/original.repo --save-filename=test.repo"
+   Then the exit code is 0
+    And file "/etc/yum.repos.d/test.repo" contents is
+        """
+        [test]
+        name=repository file
+        enabled=0
+        baseurl=http://something.com/os/
+        """
+
+
+Scenario: test "addrepo" from repofile with user defined destination file name (--save-filename=), adding the .repo extension to the destination filename
+ Given I create file "/{context.dnf.tempdir}/tmp/original.repo" with
+        """
+        [test]
+        name=repository file
+        enabled=0
+        baseurl=http://something.com/os/
+        """
+   When I execute dnf with args "config-manager addrepo --from-repofile={context.dnf.tempdir}/tmp/original.repo --save-filename=test"
+   Then the exit code is 0
+    And file "/etc/yum.repos.d/test.repo" contents is
+        """
+        [test]
+        name=repository file
+        enabled=0
+        baseurl=http://something.com/os/
+        """
+
+
+Scenario: test "addrepo" from repofile, destination directory does not exist
+ Given I delete directory "/etc/yum.repos.d/"
+    And I create file "/{context.dnf.tempdir}/tmp/test.repo" with
+        """
+        [test]
+        name=repository file
+        enabled=0
+        baseurl=http://something.com/os/
+        """
+   When I execute dnf with args "config-manager addrepo --from-repofile={context.dnf.tempdir}/tmp/test.repo"
+   Then the exit code is 1
+    And stderr contains "Directory ".*/etc/yum.repos.d" does not exist. Add "--create-missing-dir" to create missing directories"
+
+
+Scenario: test "addrepo" from repofile, destination directory does not exist, "--create-missing-dir" creates missing directory
+ Given I delete directory "/etc/yum.repos.d/"
```

Comment:
Minor: I believe that there should be an additional space before `Given:`. This is valid for multiple steps.

Reply:
I added space before `Given`